sdk/monitor/Azure.Monitor.Query/src/Models/MonitorQueryModelFactory.cs

```diff
@@ -36,5 +36,15 @@ public static LogsQueryResult LogsQueryResult(IReadOnlyList<LogsTable> allTables
             JsonElement errorJson = error.ToObjectFromJson<JsonElement>();
             return new LogsQueryResult(allTables, statisticsJson, visualizationJson, errorJson);
         }
+
+        /// <summary> Initializes a new instance of LogsTable. </summary>
+        /// <param name="name"> The name of the table. </param>
+        /// <param name="columns"> The list of columns in this table. </param>
+        /// <param name="internalRows"> The resulting rows from this query. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="name"/> or <paramref name="columns"/> is null. </exception>
+        public static LogsTable LogsTable(string name, IEnumerable<LogsTableColumn> columns, JsonElement internalRows)
```

Comment:
We need to take an `IEnumerable<LogsTableRow>` because that's the type we expose in public API.